src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -31,6 +47,71 @@
 _EXPLANATION = "explanation"
 
 
+class AsyncExecutor:
+    def __init__(
+        self,
+        generation_fn: Callable[[Any], Coroutine[Any, Any, Any]],
+        num_consumers: int = 20,
+        tqdm_bar_format: Optional[str] = None,
+    ):
+        self.generate = generation_fn
+        self.num_consumers = num_consumers
+        self.tqdm_bar_format = tqdm_bar_format
+
+        self.end_of_queue = object()
+
+    async def consumer(
+        self,
+        output: List[Any],
+        queue: asyncio.Queue[Union[object, Tuple[int, Any]]],
+        progress_bar: async_tqdm[Any],
+    ) -> None:
+        while True:
+            if (item := await queue.get()) is self.end_of_queue:
+                break
+
+            item = cast(Tuple[int, Any], item)
+            index, payload = item
+            result = await self.generate(payload)
+            output[index] = result
+            progress_bar.update()
+
+    async def producer(
+        self,
+        inputs: Sequence[Any],
+        queue: asyncio.Queue[Union[object, Tuple[int, Any]]],
+    ) -> None:
+        for index, input in enumerate(inputs):
+            await queue.put((index, input))
+        for _ in range(self.num_consumers):
+            await queue.put(self.end_of_queue)
+
+    async def submit(self, inputs: Sequence[Any]) -> List[Any]:
+        outputs = ["unset"] * len(inputs)
+        progress_bar = async_tqdm(total=len(inputs), bar_format=self.tqdm_bar_format)
+
+        queue: asyncio.Queue[Union[object, Tuple[int, Any]]] = asyncio.Queue(
+            maxsize=2 * self.num_consumers
```

Comment:
clarification: Not following the 2x consumers.

Reply:
hmm, okay don't quite follow why that requires a 2x queue at least in the current incarnation - just was thinking as a reader I don't know what 2x means